From edab77283a06cf122176b31863bbb80b4b417792 Mon Sep 17 00:00:00 2001
From: anruliu <anruliu@allwinnertech.com>
Date: Fri, 21 May 2021 09:58:17 +0800
Subject: [PATCH] add gpu configure support

Signed-off-by: anruliu <anruliu@allwinnertech.com>
---
 cmake_mgconfig.h.in |  6 ++++++
 configure.ac        | 20 ++++++++++++++++++++
 2 files changed, 26 insertions(+)

diff --git a/cmake_mgconfig.h.in b/cmake_mgconfig.h.in
index 8deedea..4a32c1a 100644
--- a/cmake_mgconfig.h.in
+++ b/cmake_mgconfig.h.in
@@ -562,6 +562,12 @@
 /* Define if support G2D Rotate */
 #cmakedefine _MGIMAGE_G2D_ROTATE    1
 
+/* Define if support GPU */
+#cmakedefine _MGRM_GPU          1
+
+/* Define if support GPU Rotate */
+#cmakedefine _MGIMAGE_GPU_ROTATE    1
+
 /* ******* */
 /* USE_OWN */
 /* ******* */
diff --git a/configure.ac b/configure.ac
index 6d08c3b..050ef90 100644
--- a/configure.ac
+++ b/configure.ac
@@ -212,6 +212,8 @@ build_jpg_support="yes"
 build_png_support="yes"
 build_g2d_support="no"
 build_g2drotate_support="no"
+build_gpu_support="no"
+build_gpurotate_support="no"
 build_pcx_support="no"
 build_lbm_support="no"
 build_tga_support="no"
@@ -706,6 +708,14 @@ AC_ARG_ENABLE(g2drotatesupport,
 [  --enable-g2drotatesupport      include G2D Rotate support <default=no>],
 build_g2drotate_support=$enableval)
 
+AC_ARG_ENABLE(gpusupport,
+[  --enable-gpusupport      include GPU support <default=no>],
+build_gpu_support=$enableval)
+
+AC_ARG_ENABLE(gpurotatesupport,
+[  --enable-gpurotatesupport      include GPU Rotate support <default=no>],
+build_gpurotate_support=$enableval)
+
 AC_ARG_ENABLE(menu,
 [  --enable-menu            include menu support <default=yes>],
 build_menu=$enableval)
@@ -2317,6 +2327,16 @@ if test "x$build_g2drotate_support" = "xyes"; then
             [Define if support G2D Rotate])
 fi
 
+if test "x$build_gpu_support" = "xyes"; then
+    AC_DEFINE(_MGIMAGE_GPU, 1,
+            [Define if support GPU])
+fi
+
+if test "x$build_gpurotate_support" = "xyes"; then
+    AC_DEFINE(_MGIMAGE_GPU_ROTATE, 1,
+            [Define if support GPU Rotate])
+fi
+
 if test "x$build_menu" = "xyes"; then
     AC_DEFINE(_MGHAVE_MENU, 1,
             [Define if support menu])
-- 
2.29.0

