From eead100957eaccd5f24ec5ba2257a98308237419 Mon Sep 17 00:00:00 2001
From: hongjiasen <hongjiasen@Allwinnertech.com>
Date: Fri, 24 Dec 2021 10:17:04 +0800
Subject: [PATCH] bluealsa: fix a2dp src start Couldn't create data buffers

Signed-off-by: hongjiasen <hongjiasen@Allwinnertech.com>
---
 src/io.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/io.c b/src/io.c
index a2788a2..d73025d 100644
--- a/src/io.c
+++ b/src/io.c
@@ -353,6 +353,16 @@ void *io_thread_a2dp_source_sbc(void *arg) {
 	const unsigned int channels = transport_get_channels(t);
 	const unsigned int samplerate = transport_get_sampling(t);
 
+	int ii = 1;
+	for(ii = 1 ;ii <= 100 ; ii++){
+		if(t->mtu_write <= 0){
+			usleep(100 * 1000);
+			error(" wait t->mtu_write timer number : %d",ii);
+		} else{
+			break;
+		}
+	}
+
 	/* Writing MTU should be big enough to contain RTP header, SBC payload
 	 * header and at least one SBC frame. In general, there is no constraint
 	 * for the MTU value, but the speed might suffer significantly. */
-- 
2.29.0

