Index: ntfs-3g-2015.3.14-1-fuseint/libntfs-3g/ioctl.c
===================================================================
--- ntfs-3g-2015.3.14-1-fuseint.orig/libntfs-3g/ioctl.c
+++ ntfs-3g-2015.3.14-1-fuseint/libntfs-3g/ioctl.c
@@ -62,6 +62,10 @@
 #include <sys/stat.h>
 #endif
 
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
+
 #ifdef HAVE_LINUX_FS_H
 #include <linux/fs.h>
 #endif
Index: ntfs-3g-2015.3.14-1-fuseint/src/lowntfs-3g.c
===================================================================
--- ntfs-3g-2015.3.14-1-fuseint.orig/src/lowntfs-3g.c
+++ ntfs-3g-2015.3.14-1-fuseint/src/lowntfs-3g.c
@@ -77,6 +77,10 @@
 #include <sys/mkdev.h>
 #endif
 
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
+
 #if defined(__APPLE__) || defined(__DARWIN__)
 #include <sys/dirent.h>
 #elif defined(__sun) && defined (__SVR4)
Index: ntfs-3g-2015.3.14-1-fuseint/src/ntfs-3g.c
===================================================================
--- ntfs-3g-2015.3.14-1-fuseint.orig/src/ntfs-3g.c
+++ ntfs-3g-2015.3.14-1-fuseint/src/ntfs-3g.c
@@ -76,6 +76,10 @@
 #include <sys/mkdev.h>
 #endif
 
+#ifdef HAVE_SYS_SYSMACROS_H
+#include <sys/sysmacros.h>
+#endif
+
 #if defined(__APPLE__) || defined(__DARWIN__)
 #include <sys/dirent.h>
 #elif defined(__sun) && defined (__SVR4)
